.github/workflows/pythonapp-gpu.yml

```diff
@@ -20,46 +21,35 @@ jobs:
     strategy:
       matrix:
         environment:
-          - "PT19+CUDA114"
-          - "PT17+CUDA102"
           - "PT18+CUDA102"
-          - "PT18+CUDA112"
+          - "PT19+CUDA114"
+          - "PT110+CUDA111"
           - "PT112+CUDA118"
-          - "PT110+CUDA102"
-          - "PT112+CUDA102"
+          - "PT113+CUDA116"
         include:
           # https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes
-          - environment: PT17+CUDA102
-            pytorch: "torch==1.7.1 torchvision==0.8.2"
-            base: "nvcr.io/nvidia/cuda:10.2-devel-ubuntu18.04"
           - environment: PT18+CUDA102
             # pytorch 1.8.2 LTS
             pytorch: "torch==1.8.2 torchvision==0.9.2 --extra-index-url https://download.pytorch.org/whl/lts/1.8/cu102"
             base: "nvcr.io/nvidia/cuda:10.2-devel-ubuntu18.04"
-          - environment: PT18+CUDA112
-            # we explicitly set pytorch to -h to avoid pip install error
-            # 21.03: 1.9.0a0+df837d0
-            pytorch: "-h"
-            base: "nvcr.io/nvidia/pytorch:21.03-py3"
           - environment: PT19+CUDA114
             # we explicitly set pytorch to -h to avoid pip install error
             # 21.10: 1.10.0a0+0aef44c
             pytorch: "-h"
             base: "nvcr.io/nvidia/pytorch:21.10-py3"
+          - environment: PT110+CUDA111
+            pytorch: "torch==1.10.2 torchvision==0.11.3 --extra-index-url https://download.pytorch.org/whl/cu111"
+            base: "nvcr.io/nvidia/cuda:11.1.1-devel-ubuntu18.04"
           - environment: PT112+CUDA118
-            # we explicitly set pytorch to -h to avoid pip install error
             # 22.09: 1.13.0a0+d0d6b1f
-            pytorch: "-h"
+            pytorch: "-h"  # we explicitly set pytorch to -h to avoid pip install error
             base: "nvcr.io/nvidia/pytorch:22.09-py3"
```

Comment:
Can we only use pytorch or cuda images. It could be confusing using both.

Reply:
thanks Yanxuan, pytorch docker for example `22.09` is based on pytorch `1.13.0a0+d0d6b1f` which is not exactly the same as pip install pytorch==1.12 or 1.13, so it's necessary to test with these base images. I updated the option names to make it clear